### PR TITLE
Update parsing to handle amount out of range 

### LINKED ISSUE #195: [PE-D][Tester B] Handling of very large transaction inputs
I called this command, `addt 1 d/buy raw materials amt/9999999999999999999999999999999999999999999999999999999999999999999999999
999999999999999999999999999 o/Company ABC dt/2024-10-16` and your program, basically broke, showing the screen below. I would suggest declaring a maximum/minimum transaction amount in your user guide :)

Screenshot:

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: hovanleong/ped#4

### PR DESCRIPTION
**Pull request by kngys**
Previously, amount out of range would cause issues with json file loading.

Fixes #195, #246

image
image

Amounts out of range result in error message.

### CODE REVIEW

**Comment by kngys** [comment]
@beginner-web-developer which issue is about the json file being unable to load because of amount too large

**Comment by beginner-web-developer** [comment]
Issue #246

**Comment by kngys** [comment]
> Issue #246

Can I clarify how does #195 cause the app to crash?

File: src/main/java/seedu/address/model/client/Transaction.java [hunk 1]
```diff
@@ -40,6 +42,19 @@ public static boolean isValidAmount(String test) {
...
+     */
+    public static void verifyAmountIsWithinRange(double amount) throws ParseException {
+        double minAmount = -Math.pow(10, 9);
+        double maxAmount = Math.pow(10, 9);
+
+        if (amount < minAmount || amount > maxAmount) {
+            throw new ParseException(MESSAGE_AMOUNT_OUT_OF_RANGE);
+        }
+
+    }
```
Comment (reginateh) [inline src/main/java/seedu/address/model/client/Transaction.java:56]: I feel like this method can be put in `ParserUtil`

Comment (kngys) [reply src/main/java/seedu/address/model/client/Transaction.java]: Ok I moved the method over

File: src/main/java/seedu/address/logic/Messages.java [hunk 2]
```diff
@@ -53,8 +53,12 @@ public class Messages {
     public static final String MESSAGE_EMPTY_TRANSACTION_LIST =
             "Invalid command: The current transaction list is empty.";
 
+    public static final String MESSAGE_AMOUNT_OUT_OF_RANGE =
+            "Amount should be between -1 billion (-1000000000) to 1 billion (1000000000)!";
```
Comment (trpmy) [inline src/main/java/seedu/address/logic/Messages.java:57]: The ! seems a little enthusiastic but that's alright

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -171,9 +172,24 @@ public static Double parseAmount(String amountStr) throws ParseException {
...
+
         return amount;
     }
 
+    /**
+     * Throws a {@code ParseException} if the amount is out of range.
+     */
+    public static void verifyAmountIsWithinRange(double amount) throws ParseException {
+        double minAmount = -Math.pow(10, 9);
+        double maxAmount = Math.pow(10, 9);
```
Comment (trpmy) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:186]: Technically fine but better practice might've been to make them a constant 1000000000

**Review by trpmy — APPROVED** [review]

**Pull request merged**